djangomodelimport/parsers.py

```diff
@@ -18,26 +18,36 @@ def parse(self, data):
 
 
 class TablibCSVImportParser(ImportParser):
-    def parse(self, data):
-        dataset = tablib.Dataset()
-        dataset.csv = data
-
+    def get_soft_headings(self):
+        # Soft headings are used to provide similar heading suggestions
+        # and look like this: {the field name: [list of other possible names] }
+        # eg.
+        #
+        # soft_headings = {
+        #   'type': ['Asset Type'],
+        # }
+        header_map = {}
         if hasattr(self.modelvalidator, 'ImporterMeta'):
-            # has additional importer config defined
             if hasattr(self.modelvalidator.ImporterMeta, 'soft_headings'):
                 importer_softheadings = self.modelvalidator.ImporterMeta.soft_headings
 
-                # rename headers according to a soft_headings mapping dict which renames
-                # columns to a consistent value
-                # header mapping looks like this
-                # soft_headings = {
-                #   'colnewname': ['fromname1', 'fromname2']
-                # }
                 for renameto in importer_softheadings:  # new column name
                     for renamefrom in importer_softheadings[renameto]:  # old column name
-                        for idx, header in enumerate(dataset.headers):  # replace it in headers if found
-                            if header.strip() == renamefrom.strip():
-                                dataset.headers[idx] = renameto
+                        header_map[renamefrom.lower()] = renameto.lower()
```

Comment:
are you sure you want to do this? I left it without a .lower because then the caller would have more control over how he wants the columns to look like.

Reply:
This is just for import -- won't make a difference? --aidan